GitUI/UserControls/RevisionGrid/RevisionDataGridView.cs

```diff
@@ -178,11 +177,21 @@ internal Font NormalFont
 
         internal AuthorRevisionHighlighting AuthorHighlighting { get; set; }
 
-        // Contains the object Id's that will be selected as soon as they are loaded.
-        public HashSet<ObjectId> ToBeSelectedObjectIds { get; set; } = new HashSet<ObjectId>();
+        // Contains the object Id's that will be selected as soon as all of them have been loaded.
+        // The object Id's are in the order in which they were originally selected.
+        public List<ObjectId> ToBeSelectedObjectIds { get; set; } = new List<ObjectId>();
```

Comment:
```suggestion
        public IReadOnlyList<ObjectId> ToBeSelectedObjectIds { get; set; } = new List<ObjectId>();
```

Reply:
Fails to compile with this change

Reply:
That was my fault.

RussKie suggested turning `ToBeSelectedObjectIds` from a `List` to an `IReadOnlyList`. That makes sense, since `ToBeSelectedObjectIds` contains a fixed list of commits that are to be selected, so I accepted the change.

However, I overlooked the fact that `IndexOf(T)` is a method of `List<T>`. For an `IReadOnlyList`, the extension method `IndexOf` on `IEnumerable<T>` takes a lambda instead of an object, as the error message indicates.